Only show mismatched diagnostics in case of error.

manfred-brands opened the pull request:
Fixes #256

JohanLarsson commented:
Very nice, the error text was not great for this.

Do you have more pull requests in the pipe or should I release an updated package?

manfred-brands commented:
This was the result of adding it to an Analyzer for work, which used to use MSTest with lots of .UseLocation calls.
Since I hated counting lines, I moved the tests to use Gu.Roslyn.Assert, but I forgot one or two ↓ markers.
So it showed me mismatch with two long lists of a dozen lines.
After this change it showed 'Matched: 11, Missing: ...'

Wait with a new package until I moved the nunit.analyzers over in case I need something there.

Can you point me where in the Gu.Roslyn.Assert.Analyzers I need to look to raise an issue if there are no ↓ markers in code passed to `Roslyn.Suppressed`?

JohanLarsson commented:
> Can you point me where in the Gu.Roslyn.Assert.Analyzers I need to look to raise an issue if there are no ↓ markers in code passed to Roslyn.Suppressed?

I don't remember, maybe easiest to debug an existing test?
